portable/GCC/ARM_CM3/port.c

```diff
@@ -265,10 +265,6 @@ static void prvPortStartFirstTask( void )
  */
 BaseType_t xPortStartScheduler( void )
 {
-    /* configMAX_SYSCALL_INTERRUPT_PRIORITY must not be set to 0.
```

Comment:
the configASSERT( configMAX_SYSCALL_INTERRUPT_PRIORITY ); was previously outside "#if ( configASSERT_DEFINED == 1 )" check. With this change, the assert will be moved to inside the check . Just trying to understand the reason why it was outside previously and will there be an impact after it's moved inside now.

Reply:
It uses the ucMaxSysCallPriority, which is only in scope inside the #if to prevent it causing compiler warnings when configASSERT() isn't used.

Reply:
Yeah, in both cases it's only doing anything if you have a configASSERT. The expressions passed to the new asserts are in scope.